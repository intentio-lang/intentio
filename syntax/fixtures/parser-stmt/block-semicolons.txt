if a {
    ;;if b {
        c;;;
    } else {
        d;
    };

    while b {
        {}
        d;;
        {}
    }
}
