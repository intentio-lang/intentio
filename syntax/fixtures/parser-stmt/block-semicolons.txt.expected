[
  {
    "_stmtAnn": [],
    "_stmtKind": {
      "contents": {
        "_exprAnn": [],
        "_exprKind": {
          "contents": [
            {
              "_exprAnn": [],
              "_exprKind": {
                "contents": {
                  "contents": {
                    "_scopeIdAnn": [],
                    "_scopeIdSourcePos": [
                      "test.ieo",
                      0,
                      3
                    ],
                    "_unScopeId": "a"
                  },
                  "tag": "ScopeId'"
                },
                "tag": "IdExpr"
              },
              "_exprSourcePos": [
                "test.ieo",
                0,
                3
              ]
            },
            {
              "_blockAnn": [],
              "_blockSourcePos": [
                "test.ieo",
                0,
                5
              ],
              "_blockStmts": [
                {
                  "_stmtAnn": [],
                  "_stmtKind": {
                    "contents": {
                      "_exprAnn": [],
                      "_exprKind": {
                        "contents": [
                          {
                            "_exprAnn": [],
                            "_exprKind": {
                              "contents": {
                                "contents": {
                                  "_scopeIdAnn": [],
                                  "_scopeIdSourcePos": [
                                    "test.ieo",
                                    1,
                                    9
                                  ],
                                  "_unScopeId": "b"
                                },
                                "tag": "ScopeId'"
                              },
                              "tag": "IdExpr"
                            },
                            "_exprSourcePos": [
                              "test.ieo",
                              1,
                              9
                            ]
                          },
                          {
                            "_blockAnn": [],
                            "_blockSourcePos": [
                              "test.ieo",
                              1,
                              11
                            ],
                            "_blockStmts": [
                              {
                                "_stmtAnn": [],
                                "_stmtKind": {
                                  "contents": {
                                    "_exprAnn": [],
                                    "_exprKind": {
                                      "contents": {
                                        "contents": {
                                          "_scopeIdAnn": [],
                                          "_scopeIdSourcePos": [
                                            "test.ieo",
                                            2,
                                            8
                                          ],
                                          "_unScopeId": "c"
                                        },
                                        "tag": "ScopeId'"
                                      },
                                      "tag": "IdExpr"
                                    },
                                    "_exprSourcePos": [
                                      "test.ieo",
                                      2,
                                      8
                                    ]
                                  },
                                  "tag": "ExprStmt"
                                },
                                "_stmtSourcePos": [
                                  "test.ieo",
                                  2,
                                  8
                                ]
                              }
                            ]
                          },
                          {
                            "_blockAnn": [],
                            "_blockSourcePos": [
                              "test.ieo",
                              3,
                              11
                            ],
                            "_blockStmts": [
                              {
                                "_stmtAnn": [],
                                "_stmtKind": {
                                  "contents": {
                                    "_exprAnn": [],
                                    "_exprKind": {
                                      "contents": {
                                        "contents": {
                                          "_scopeIdAnn": [],
                                          "_scopeIdSourcePos": [
                                            "test.ieo",
                                            4,
                                            8
                                          ],
                                          "_unScopeId": "d"
                                        },
                                        "tag": "ScopeId'"
                                      },
                                      "tag": "IdExpr"
                                    },
                                    "_exprSourcePos": [
                                      "test.ieo",
                                      4,
                                      8
                                    ]
                                  },
                                  "tag": "ExprStmt"
                                },
                                "_stmtSourcePos": [
                                  "test.ieo",
                                  4,
                                  8
                                ]
                              }
                            ]
                          }
                        ],
                        "tag": "IfExpr"
                      },
                      "_exprSourcePos": [
                        "test.ieo",
                        1,
                        6
                      ]
                    },
                    "tag": "ExprStmt"
                  },
                  "_stmtSourcePos": [
                    "test.ieo",
                    1,
                    6
                  ]
                },
                {
                  "_stmtAnn": [],
                  "_stmtKind": {
                    "contents": {
                      "_exprAnn": [],
                      "_exprKind": {
                        "contents": [
                          {
                            "_exprAnn": [],
                            "_exprKind": {
                              "contents": {
                                "contents": {
                                  "_scopeIdAnn": [],
                                  "_scopeIdSourcePos": [
                                    "test.ieo",
                                    7,
                                    10
                                  ],
                                  "_unScopeId": "b"
                                },
                                "tag": "ScopeId'"
                              },
                              "tag": "IdExpr"
                            },
                            "_exprSourcePos": [
                              "test.ieo",
                              7,
                              10
                            ]
                          },
                          {
                            "_blockAnn": [],
                            "_blockSourcePos": [
                              "test.ieo",
                              7,
                              12
                            ],
                            "_blockStmts": [
                              {
                                "_stmtAnn": [],
                                "_stmtKind": {
                                  "contents": {
                                    "_exprAnn": [],
                                    "_exprKind": {
                                      "contents": {
                                        "_blockAnn": [],
                                        "_blockSourcePos": [
                                          "test.ieo",
                                          8,
                                          8
                                        ],
                                        "_blockStmts": []
                                      },
                                      "tag": "BlockExpr"
                                    },
                                    "_exprSourcePos": [
                                      "test.ieo",
                                      8,
                                      8
                                    ]
                                  },
                                  "tag": "ExprStmt"
                                },
                                "_stmtSourcePos": [
                                  "test.ieo",
                                  8,
                                  8
                                ]
                              },
                              {
                                "_stmtAnn": [],
                                "_stmtKind": {
                                  "contents": {
                                    "_exprAnn": [],
                                    "_exprKind": {
                                      "contents": {
                                        "contents": {
                                          "_scopeIdAnn": [],
                                          "_scopeIdSourcePos": [
                                            "test.ieo",
                                            9,
                                            8
                                          ],
                                          "_unScopeId": "d"
                                        },
                                        "tag": "ScopeId'"
                                      },
                                      "tag": "IdExpr"
                                    },
                                    "_exprSourcePos": [
                                      "test.ieo",
                                      9,
                                      8
                                    ]
                                  },
                                  "tag": "ExprStmt"
                                },
                                "_stmtSourcePos": [
                                  "test.ieo",
                                  9,
                                  8
                                ]
                              },
                              {
                                "_stmtAnn": [],
                                "_stmtKind": {
                                  "contents": {
                                    "_exprAnn": [],
                                    "_exprKind": {
                                      "contents": {
                                        "_blockAnn": [],
                                        "_blockSourcePos": [
                                          "test.ieo",
                                          10,
                                          8
                                        ],
                                        "_blockStmts": []
                                      },
                                      "tag": "BlockExpr"
                                    },
                                    "_exprSourcePos": [
                                      "test.ieo",
                                      10,
                                      8
                                    ]
                                  },
                                  "tag": "ExprStmt"
                                },
                                "_stmtSourcePos": [
                                  "test.ieo",
                                  10,
                                  8
                                ]
                              }
                            ]
                          }
                        ],
                        "tag": "WhileExpr"
                      },
                      "_exprSourcePos": [
                        "test.ieo",
                        7,
                        4
                      ]
                    },
                    "tag": "ExprStmt"
                  },
                  "_stmtSourcePos": [
                    "test.ieo",
                    7,
                    4
                  ]
                }
              ]
            },
            null
          ],
          "tag": "IfExpr"
        },
        "_exprSourcePos": [
          "test.ieo",
          0,
          0
        ]
      },
      "tag": "ExprStmt"
    },
    "_stmtSourcePos": [
      "test.ieo",
      0,
      0
    ]
  }
]
